### ISSUE TITLE
tf.idf calculation

### ISSUE THREAD

**Issue by rm-bwana** [RUG-IDS/Assignments#26]
If I could ask for some clarification regarding the 2nd exercise 1st question (the calculation of TF.IDF).

I have run the demos provided by @mircealungu but when I try it with one of the recommended alternate corpus it returns an error saying the CorpusReader is not iterable. Am I correct in assuming that some work needs to be done on the corpus before TF.IDF can be calculated?

What I have tried so far:
- Calculating TF.IDF as in the demo but with the individual documents in the corpus. This returns results most of the times (the other times it runs out of memory) but that defeats the point as IDF is only accurate when considering multiple documents at the same time unless there is some way of linking the outputs all back together after the calculation that I am unaware of.

I am treating the 2nd exercise as a standalone exercise or is it required to use the output of exercise 1 in some way?

**Comment by mircealungu** [comment]
You are right @rm-bwana that you need to process a bit the corpus before applying TF.IDF on it. 

If you read about the TfidfVectorizer you'll see that it takes as input to the fit_transform function an array of strings, not a corpus. Something like: 

```
shake_corpus = []
texts = ['whitman-leaves.txt', 'shakespeare-caesar.txt', 'shakespeare-hamlet.txt', 'shakespeare-macbeth.txt']
texts = gutenberg.fileids()
for text in texts:
    shake_corpus.append(gutenberg.raw(text))

```

The second exercise is indeed stand-alone.
